### ISSUE TITLE
Unable to run on ubuntu 18.04

### ISSUE THREAD

**Issue by GodOfPlutonium** [Witko/nvidia-xrun#70]
I have installed this on ubuntu 18.04 and it did not work. Looking at the log files, this was most likely becuase of several module paths being wrong. As i corrected module paths , gradualy, more and more code ran but the system never functioned. I have attached log files, of the initial unchanged path run, the second run, where i changed the first two paths, and the final run that ive done so far
[first.log ](https://github.com/Witko/nvidia-xrun/files/2296217/first.log)
[second.log](https://github.com/Witko/nvidia-xrun/files/2296218/second.log)
[thrid.log](https://github.com/Witko/nvidia-xrun/files/2296219/thrid.log)

here is the current edited path files that I have been using.

`Section "Files"
  ModulePath "/usr/lib/x86_64-linux-gnu"
  ModulePath "/usr/lib/i386-linux-gnu"
  ModulePath "/usr/lib/xorg/modules/drivers"
  ModulePath "/usr/lib/xorg/modules"
  ModulePath "/usr/lib64/nvidia/xorg/modules"
  ModulePath "/usr/lib64/nvidia/xorg"
  ModulePath "/usr/lib64/xorg/modules"
EndSection
`
Incidentally, is there any documentation on what each path is supposed to point to?

**Comment by blazessdd** [comment]
i have just added
ModulePath "/usr/lib/nvidia/xorg/modules"

ModulePath "/usr/lib/nvidia/xorg"

ModulePath "/usr/lib/xorg/modules"
and it works (debian but should work on ubuntu too

**Comment by BobbyWibowo** [comment]
Yeah, there is no `/usr/lib64` in my Ubuntu 18.04 installation. I assume all of those are in `/usr/lib` considering it's a 64-bit installation and there's a separate `/usr/lib32` for 32-bit libraries.

**Comment by BobbyWibowo** [comment]
For Ubuntu, it seems NVIDIA's proprietary driver will install to `/usr/lib/x86_64-linux-gnu/nvidia` instead.
After changing the paths in `/etc/X11/nvidia-xorg.conf` and `/etc/X11/xinit/nvidia-xinitrc` accordingly, unfortunately I'm still unable to get it to work.
For the latter file, I had to edit the path in this line specifically:
https://github.com/Witko/nvidia-xrun/blob/b8bd592d4727852dfd2cca8c23fe7e2312a93dc1/nvidia-xinitrc#L26
Basically from `/usr/lib64/nvidia/` to `/usr/lib/x86_64-linux-gnu/nvidia/`.

Here's a gist of my latest Xorg log file: https://gist.github.com/BobbyWibowo/d8a5db1c6ead2bafdd8587ce5d7c30bb.

Another thing to note, after it fails to do its thing once (the log is above), all tty's will be spammed with these errors roughly every second: https://i.fiery.me/yfZC.mp4 (~7s video).

**Comment by BobbyWibowo** [comment]
Ok, purged out bumblebee and reinstalled nvidia driver. Now I got it to only output:
```
[ 99.407] xf86EnableIOPorts: failed to set IOPL for I/O (Operation not permitted)
```
when failing.
Here's the full log: https://gist.github.com/BobbyWibowo/9bfb6c91d6e17ee3f3054445862ea5e5.
There were no other `(EE)` lines, so I legit have no idea what else to attempt to fix.
Some thing that got me concerned are these lines:
```
[    99.399] (**) OutputClass "Nvidia Prime" ModulePath extended to "/x86_64-linux-gnu/nvidia/xorg,/usr/lib/x86_64-linux-gnu/xorg,/usr/lib/x86_64-linux-gnu/nvidia,/usr/lib/x86_64-linux-gnu/nvidia/xorg/modules,/usr/lib/x86_64-linux-gnu/nvidia/xorg,/usr/lib/xorg/modules"
[ 99.399] (**) OutputClass "Nvidia Prime" setting /dev/dri/card1 as PrimaryGPU
```
They sound as if there were some leftover configurations from PRIME that sorta screwed the whole things up, but I was sure I already purged out `nvidia-prime` package.

**Comment by BobbyWibowo** [comment]
Oh thanks lord the almighty, I've finally got this working very well.
The issues/speculations in [my previous reply](https://github.com/Witko/nvidia-xrun/issues/70#issuecomment-469004905) doesn't appear to be any breaking issues (or perhaps were not issues in the first place). I still see them on my X log despite everything working great.
As for the issue in [my reply before it](https://github.com/Witko/nvidia-xrun/issues/70#issuecomment-468980558), it was probably just due to me having quite dirty configurations. I also still had bumblebee installed back then (yes, I've purged out bumblebee, don't need it).

**Comment by BobbyWibowo** [comment]
@Quintasan Here you go:
`nvidia-xorg.conf`: https://gist.github.com/BobbyWibowo/a10ac8fb77e9521987f1bf543e7d580f
`nvidia-xinitrc`: https://gist.github.com/BobbyWibowo/ab6ccb09a305eb6f3f488d065be4bae1

Also, you may want to consider using these combination of `nvidia-xrun` + `nvidia-toggle` scripts:
`nvidia-xrun`: https://gist.github.com/BobbyWibowo/09de27597520bb4bb1d816c15f40f9f5
`nvidia-toggle`: https://gist.github.com/BobbyWibowo/c7f97bca11e37502be350733b2236e94
`/etc/default/nvidia-toggle`: https://gist.github.com/BobbyWibowo/b5ac9339b0e9b81bb00d3debedf71f7b
These will use `bbswitch` instead of writing to the system file interface kinda thing (the current method in this repo).
I split the parts that needed `sudo` into `nvidia-toggle` so that I can add that script only into sudoers file, to avoid entering sudo password every single time.

I also stopped using `modeset=1` flag due to some issues back then, but by now I can't recall a single thing what the issues were, lmao.
One thing I know for sure, without the flag I'll get screen tearing, but I convinced myself that it was a better trade-off than the issues that I'd get with the flag on; not that I can recall what the issues were..
I can't remember why, but for some reason I couldn't get the card to turn off when not in use without using `bbswitch`.

**Comment by Quintasan** [comment]
@BobbyWibowo unfortunately bbswitch doesn't seem to like my device, I get `no such device` when trying to `insmod bbswitch` :(.
